js/forum/src/initializers/routes.js

```diff
@@ -35,7 +35,7 @@ export default function(app) {
    */
   app.route.discussion = (discussion, near) => {
     return app.route(near && near !== 1 ? 'discussion.near' : 'discussion', {
-      id: discussion.id() + '-' + discussion.slug(),
+      id: discussion.id() + (discussion.slug().trim() ? '-' + discussion.slug() : ''),
```

Comment:
Please extract `discussion.slug()` as a variable before the `return` statement (ie. `const slug = discussion.slug()`)